Tests/scripts/validate_premium_packs.py

```diff
@@ -0,0 +1,266 @@
+    with zipfile.ZipFile(index_zip_path, 'r') as zip_obj:
+        extracted_path = zip_obj.extract(member=f"index/{INDEX_FILE_PATH}", path="./extracted-index")
+
+    logging.info(f'extracted path is now: {extracted_path}')
+
+    return f"./{extracted_path}"
+
+
+def check_and_return_index_data(index_file_path):
+    """Check index.json file inside the index.zip archive in the cloud.
+
+    Validate no missing ids are found and that all packs have a positive price.
+
+    Args:
+        index_file_path: The path to the index.json.
+
+    Returns: Dict with the index data.
+
+    """
+    with open(index_file_path, 'r') as index_file:
+        index_data = json.load(index_file)
+
+    logging.info("Found index data in index file. Checking...")
+    logging.debug(f"Index data is:\n {index_data}")
+
+    packs_list_exists = log_message_if_statement(statement=(len(index_data["packs"]) != 0),
+                                                 error_message="Found 0 packs in index file."
+                                                               "\nAborting the rest of the check.")
+    if not packs_list_exists:
+        return False
+
+    packs_are_valid = True
+    for pack in index_data["packs"]:
+        id_exists = log_message_if_statement(statement=(pack["id"] != ""),
+                                             error_message="There is a missing pack id.")
+        price_is_valid = log_message_if_statement(statement=(pack["price"] > 0),
+                                                  error_message=f"The price on the pack {pack['id']} is 0 or less")
+        if (not id_exists) or (not price_is_valid):
+            packs_are_valid = False
+
+    log_message_if_statement(statement=packs_are_valid,
+                             error_message=f"The packs in the {index_file_path} file were found invalid.",
+                             success_message=f"{index_file_path} file was found valid")
+    return packs_are_valid, index_data
+
+
+def get_paid_packs(client: demisto_client, request_timeout: int = 999999):
+    """Get premium packs from client.
+
+    Trigger an API request to demisto server.
+    Request is identical to checking the premium box through the marketplace GUI.
+
+    Args:
+        client: The demisto client to the preform request on.
+        request_timeout: Timeout of API request
+
+    Returns:
+        Dict of premium packs as found in the server.
+        Return None if no premium packs were found.
+    """
+    request_data = \
+        {
+            'page': 0,
+            'size': 50,
+            'sort':
+                [{
+                    'field': 'updated',
+                    'asc': False
+                }],
+            'general': ["generalFieldPaid"]
+        }
+
+    logging.info(f'Getting premium packs from server {client.api_client.configuration.host}:\n')
+
+    # make the pack installation request
+    response_data, status_code, _ = demisto_client.generic_request_func(client,
```

Comment:
Not sure but looking at other places where the client is used I think this can raise exception too.

If that's true wrap this in try except and log the exception